ambry-utils/src/main/java/com/github/ambry/utils/Utils.java

```diff
@@ -1026,8 +1030,14 @@ public static void shutDownExecutorService(ExecutorService executorService, long
    * @return {@code true} this cause indicates a possible early termination from the client. {@code false} otherwise.
    */
   public static boolean isPossibleClientTermination(Throwable cause) {
-    return cause instanceof IOException && (CLIENT_RESET_EXCEPTION_MSG.equals(cause.getMessage())
-        || CLIENT_BROKEN_PIPE_EXCEPTION_MSG.equals(cause.getMessage()));
+    if (cause instanceof SSLException) {
+      return SSL_ENGINE_CLOSED_EXCEPTION_MSG.equals(cause.getMessage());
```

Comment:
Actually SSLException is an IOException, so we don't have to do this if-else statement.

Reply:
That is true, it is an IOException. I will simplify this code block by just checking the message. The one thing we lose out on by doing that is if there is a rogue IOException that is not an SSLException with the same message, but that is not likely (and we may want to catch it anyways).